compiler_opt/rl/generate_bc_trajectories.py

```diff
@@ -102,3 +191,151 @@ def get_advice(self, state: time_step.TimeStep) -> np.ndarray:
+      action_spec = None
+    else:
+      obs_spec = obs_action_specs[0].observation
+      action_spec = obs_action_specs[1]
+
+    if 'reward_key' not in kwargs:
+      raise KeyError(
+          'reward_key not specified in ExplorationWorker initialization.')
+    self.reward_key = kwargs['reward_key']
+    kwargs.pop('reward_key', None)
+    self.working_dir = None
+
+    self.env = env.MLGOEnvironmentBase(
+        clang_path=clang_path,
+        task_type=mlgo_task,
+        obs_spec=obs_spec,
+        action_spec=action_spec,
+    )
+    if self.env.action_spec:
+      if self.env.action_spec.dtype != tf.int64:
+        raise TypeError(
+            f'Environment action_spec type {0} does not match tf.int64'.format(
+                self.env.action_spec.dtype))
+    self.exploration_frac = exploration_frac
+    self.max_exploration_steps = max_exploration_steps
+    self.exploration_policy_distr = exploration_policy_distr
+    logging.info('Reward key in exploration worker: %s', self.reward_key)
+
+  def compile_module(
+      self,
+      policy: Callable[[Optional[time_step.TimeStep]], np.ndarray],
+  ) -> tf.train.SequenceExample:
+    """Compiles the module with the given policy and outputs a seq. example.
+
+    Args:
+      policy: policy to compile with
+
+    Returns:
+      sequence_example: a tf.train.SequenceExample containing the trajectory
+        from compilation.
+    """
+    sequence_example = tf.train.SequenceExample()
+    curr_obs_dict = self.env.reset(self.loaded_module_spec)
+    try:
+      curr_obs = curr_obs_dict.obs
+      self._process_obs(curr_obs, sequence_example)
+      while curr_obs_dict.step_type != env.StepType.LAST:
+        timestep = self._create_timestep(curr_obs_dict)
+        action = policy(timestep)
+        add_int_feature(sequence_example, int(action), 'action')
+        curr_obs_dict = self.env.step(action)
+        curr_obs = curr_obs_dict.obs
+        if curr_obs_dict.step_type == env.StepType.LAST:
+          break
+        self._process_obs(curr_obs, sequence_example)
+    except AssertionError as e:
+      logging.error('AssertionError: %s', e)
+    horizon = len(sequence_example.feature_lists.feature_list['action'].feature)
+    self.working_dir = curr_obs_dict.working_dir
+    if horizon <= 0:
+      working_dir_head = os.path.split(self.working_dir)[0]
+      shutil.rmtree(working_dir_head)
+    if horizon <= 0:
+      raise ValueError(
+          f'Policy did not take any inlining decision for module {0}.'.format(
+              self.loaded_module_spec.name))
+    if curr_obs_dict.step_type != env.StepType.LAST:
+      raise ValueError(
+          f'Compilation loop exited at step type {0} before last step'.format(
+              curr_obs_dict.step_type))
+    native_size = curr_obs_dict.score_policy[self.reward_key]
+    native_size_list = np.float32(native_size) * np.float32(np.ones(horizon))
+    add_feature_list(sequence_example, native_size_list, 'reward')
```

Comment:
so I understand: the reward key could be `hi_mom`, but we rename it here to `reward`?

Reply:
The sequence_examples are what is going to be used for training. reward_key is the the key to be used in the environment to determine what we use as a reward, the two are somewhat independent.

Reply:
may be worth commenting that, easier for a reader to get that "it's intentional"

Reply:
See if the docstring for the function makes this more clear now.